spec/tictactoe_spec.rb

```diff
@@ -1,26 +1,123 @@
     it 'returns a single list with 9 elements' do 
-     game = TicTacToe.new
+     game = TicTacToe.new(io: FakeIO.new)
 
      expect(game.renderBoard.length).to eql 9
-    end 
-
-    it 'returns a list of all strings' do 
-     game = TicTacToe.new
+     expect(game.renderBoard.all?{ |space| space.class == String && space.length == 1 } ).to eql true 
+    end
 
-     expect(game.renderBoard.all?{ |space| space.class == String && space.length <= 1 } ).to eql true 
-    end 
   end
 
   context 'when the game starts the board is empty' do 
-    it 'sets the board to a list with empty strings' do 
-        game = TicTacToe.new
+    it 'sets the board to a list with a single space' do 
+        game = TicTacToe.new(io: FakeIO.new)
         game.newGame 
-        expect( game.renderBoard.all?{ |space| space == '' } ).to eql true
+        expect( game.renderBoard.all?{ |space| space == ' ' } ).to eql true
+    end 
+  end  
+
+ context 'when a single token is placed on an empty board' do 
+    it 'the token will be "X"' do 
+      game = TicTacToe.new(io: FakeIO.new) 
+      expect( game.currentPlayer).to eql 'X' 
+    end
+
+    it 'the specified space will place the current token on the board' do 
+      fakeIO = FakeIO.new 
+      game = TicTacToe.new(io: fakeIO)  
+      game.placeToken(5) 
+      
+      isRowOutputed(io: fakeIO, row:" |X| ") 
+    end
+
+    it 'the token can be placed anywhere on the board' do 
+      fakeIO = FakeIO.new
+      expectedRowOutput = ""
+      game = TicTacToe.new(io: fakeIO)
+
+      game.placeToken(3)
+      isRowOutputed(io: fakeIO, row:" | |X" )  
+      
+      game.newGame
+      fakeIO.stdout = []
+      game.placeToken(5) 
+      isRowOutputed(io: fakeIO, row: " |X| " )
+      
+      
+      game.newGame
+      fakeIO.stdout = []  
+      game.placeToken(7)
+      isRowOutputed(io: fakeIO, row: "X| | ") 
+    end 
+    
+    it 'renders the board after a token is placed' do 
+      fakeIO = FakeIO.new
+      game = TicTacToe.new(io: fakeIO)
+
+       game.placeToken(3)
+       expectedBoardOutput = [" | |X", "------", " | | ", "------", " | | "]
+
+       expect( fakeIO.stdout ).to eql expectedBoardOutput
+    end   
+  end  
+
+  context 'when the user enters a desired position' do 
+    it 'renders the board with their token' do
+      desiredPosition = 4 
+   
+      userInput = StringIO.new(desiredPosition.to_s) 
+      fakeIO = FakeIO.new( stdin: userInput )
+      game = TicTacToe.new(io: fakeIO )
+    
+      game.playerTurn
+    
+      isRowOutputed(io: fakeIO, row: "X| | ") 
+    end 
+  end 
+
+ context 'when the user tries to place a token in a free space' do 
+    it 'renders the board with their token in the specified space' do 
+      desiredPosition = 4 
+   
+      userInput = StringIO.new(desiredPosition.to_s) 
+      fakeIO = FakeIO.new( stdin: userInput )
+      game = TicTacToe.new(io: fakeIO )
+    
+      expect(game.renderBoard[ desiredPosition - 1] ).to eql ' '
```

Comment:
What does `desiredPosition - 1` mean? Is this something the client of `TicTacToe` should know about?

Reply:
I looked back at the code and realized wasn't subtracting from 1 isn't needed. I'll remove it and keep it as `[desiredPosition]` or rather `[desired_position]`.